inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -984,25 +984,20 @@ def _assign_institution_ref_to_affiliation(matched_affiliation):
         matched_affiliation["record"] = institution_reference
 
 
-def _assign_matched_affiliations_to_author(
-    author_affiliations, matched_author_affiliations
-):
-    matched_affiliations_to_assign = []
-    for matched_author_affiliation in matched_author_affiliations:
-        if "record" not in matched_author_affiliation:
+def _assign_institution_reference_to_incomplete_affiliations(author_affiliations):
+    for affiliation in author_affiliations:
+        if "record" not in affiliation:
             _assign_institution_ref_to_affiliation(
-                matched_author_affiliation
+                affiliation
             )
-        matched_affiliations_to_assign.append(matched_author_affiliation)
-    author_affiliations.extend(matched_affiliations_to_assign)
-    return matched_affiliations_to_assign
 
 
 def normalize_affiliations(obj, eng):
     matched_affiliations = {}
     for author in obj.data.get("authors", []):
         author_affiliations = author.get("affiliations", [])
         if author_affiliations:
+            _assign_institution_reference_to_incomplete_affiliations(author_affiliations)
```

Comment:
this is doing one search per author, instead of one search per unique affiliation. Also you're doing it only for core records now. I think it would be cleaner to take it out into a different workflow task. If you run it after normalization, you don't need to call it here at all (on L1009 currently).